examples/nixnet/can-signal-single-point-output.py

```diff
@@ -0,0 +1,125 @@
+import grpc
+import nixnet_pb2 as nixnet_types
+import nixnet_pb2_grpc as grpc_nixnet
+
+SERVER_ADDRESS = "localhost"
+SERVER_PORT = "31763"
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    SERVER_ADDRESS = sys.argv[1]
+if len(sys.argv) >= 3:
+    SERVER_PORT = sys.argv[2]
+
+# Parameters
+INTERFACE = "CAN1"
+DATABASE = "NIXNET_example"
+CLUSTER = "CAN_Cluster"
+SIGNAL_LIST = "CANEventSignal1,CANEventSignal2"
+NUM_SIGNALS = 2
+SUCCESS = 0
+
+
+def check_for_error(status):
+    """Raise an exception if the status indicates an error."""
+    if status != 0:
+        error_message_response = client.StatusToString(
+            nixnet_types.StatusToStringRequest(status_id=status)
+        )
+        raise Exception(error_message_response.status_description)
+
+
+i = 0
+value_buffer = [0.0] * NUM_SIGNALS
+
+# Create the communication channel for the remote host and create connections to the NI-XNET and
+# session services.
+channel = grpc.insecure_channel(f"{SERVER_ADDRESS}:{SERVER_PORT}")
+client = grpc_nixnet.NiXnetStub(channel)
+
+# Display parameters that will be used for the example.
+print("Interface: " + INTERFACE, "Database: " + DATABASE, "Signal List: " + SIGNAL_LIST, sep="\n")
+
+try:
+    # Create an XNET session in SignalOutSinglePoint mode
+    create_session_response = client.CreateSession(
+        nixnet_types.CreateSessionRequest(
+            database_name=DATABASE,
+            cluster_name=CLUSTER,
+            list=SIGNAL_LIST,
+            interface=INTERFACE,
+            mode=nixnet_types.CREATE_SESSION_MODE_MODE_SIGNAL_OUT_SINGLE_POINT,
+        )
+    )
+    check_for_error(create_session_response.status)
+
+    session_reference = create_session_response.session_ref
+
+    print("Session Created Successfully.\n")
+    print("\nPress any key to transmit new signal values or q to quit\n")
+
+    while not (getch.getch()).decode("UTF-8") == "q":
+        value_buffer[0] = float(i)
+        value_buffer[1] = float(i * 10)
+
+        # Update the signal data
+        write_signal_response = client.WriteSignalSinglePoint(
+            nixnet_types.WriteSignalSinglePointRequest(
+                session_ref=session_reference, value_buffer=value_buffer
+            )
+        )
+        check_for_error(write_signal_response.status)
+
+        print("Signals sent:")
+        print(f"Signal 1: {value_buffer[0]}")
+        print(f"Signal 2: {value_buffer[1]}", end="\n\n")
+        i = i + 1 if i < 10 else 0
+
+except grpc.RpcError as rpc_error:
+    error_message = rpc_error.details()
+    if rpc_error.code() == grpc.StatusCode.UNAVAILABLE:
+        error_message = f"Failed to connect to server on {SERVER_ADDRESS}:{SERVER_PORT}"
+    elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
+        error_message = (
+            "The operation is not implemented or is not supported/enabled in this service"
+        )
+    print(f"{error_message}")
+
+finally:
+    if "session_reference" in vars() and session_reference.id != 0:
```

Comment:
I'm pretty sure that this can all just be `if session_reference:`.

Also make sure that you include some error cases in your testing (and include that in the PR notes).

Reply:
Sure